airbyte-cdk/python/airbyte_cdk/sources/lcc/retrievers/simple_retriever.py

```diff
@@ -0,0 +1,236 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Union
+
+import requests
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.lcc.extractors.extractor import Extractor
+from airbyte_cdk.sources.lcc.requesters.paginators.paginator import Paginator
+from airbyte_cdk.sources.lcc.requesters.requester import Requester
+from airbyte_cdk.sources.lcc.retrievers.retriever import Retriever
+from airbyte_cdk.sources.lcc.states.state import State
+from airbyte_cdk.sources.lcc.stream_slicers.StreamSlicer import StreamSlicer
+from airbyte_cdk.sources.streams.http import HttpStream
+
+
+class SimpleRetriever(Retriever, HttpStream):
```

Comment:
I think I'm a bit confused on why this is called a retriever and not a ConfigurableHttpStream since it seems to be exactly the latter

Reply:
the reason for not calling it a stream is we might want different types of retrieval flavors for a configurable stream (the most obvious addition being a retriever with a threadpool).

The only reason SimpleRetriever is inheriting `HttpStream` is because extracting the logic for `read_records`, `_send_request`, `_send`, and `_create_prepared_request` out of the `HttpStream` and reusing it from a `Requester` is a fairly heavy lift